wagtail/query.py

```diff
@@ -179,6 +185,50 @@ def is_specific(self):
             (SpecificIterable, DeferredSpecificIterable),
         )
 
+    def select_related(self, *fields, for_specific_subqueries: bool = False):
+        """
+        Return a new QuerySet instance that will select related objects.
+
+        If fields are specified, they must be ForeignKey fields and only those
+        related objects are included in the selection.
+
+        If select_related(None) is called, clear the list.
+
+        When called with `for_specific_subqueries=True`, the selected fields
+        will be combined
+        """
+
+        if not for_specific_subqueries:
+            return super().select_related(*fields)
+        clone = self._chain()
+        if not fields:
+            raise ValueError(
+                "'fields' must be specified when calling select_related() with for_specific_subqueries=True"
+            )
+        if fields == (None,):
+            clone._specific_select_related_fields = ()
+        else:
+            clone._specific_select_related_fields = (
+                self._specific_select_related_fields + fields
+            )
+        return clone
+
+    def prefetch_related(self, *lookups, for_specific_subqueries: bool = False):
+        if not for_specific_subqueries:
+            return super().prefetch_related(*lookups)
+        clone = self._chain()
+        if not lookups:
+            raise ValueError(
+                "'lookups' must be provided when calling prefetch_related() with for_specific_subqueries=True"
+            )
+        if lookups == (None,):
+            clone._specific_prefetch_related_lookups = ()
+        else:
+            clone._specific_prefetch_related_lookups = (
+                self._specific_prefetch_related_lookups + lookups
+            )
```

Comment:
```suggestion
            clone._specific_prefetch_related_lookups += lookups
```

Reply:
@RealOrangeOne Does this work for tuples?